src/codegen/compile.ml

```diff
@@ -5881,7 +5981,7 @@ module FuncDec = struct
         (* Deserialize and look up closure argument *)
         Serialization.deserialize env Type.[Prim Nat32] ^^
         BoxedSmallWord.unbox env ^^
-        ClosureTable.recall env ^^
+        ClosureTable.peek_future env ^^
```

Comment:
Just to clarify, if the future traps, when does the future itself get recalled? Is the answer as part of the reject handler of the self-call?

Reply:
If the future traps ("on the other side") then the reject handler will clean up. If that fails, the cleanup-callback is the last resort.